src/lib/css-tag.ts

```diff
@@ -0,0 +1,52 @@
+/**
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+export const supportsAdoptedStyleSheets = ('adoptedStyleSheets' in Document.prototype);
+
+class LiteralString {
+
+  value: string;
+
+  constructor(value: string) {
+    this.value = value.toString();
+  }
+
+  toString() {
+    return this.value;
+  }
+}
+
+const literalValue = (value: LiteralString) => {
+  if (value instanceof LiteralString) {
+    return value.value;
+  } else {
+    throw new Error(
+        `non-literal value passed to 'css' function: ${value}`
+    );
+  }
+};
+
+export const css = (strings: string[], ...values: any[]) => {
```

Comment:
I think we should consider having this return an object with an optional `.styleSheet` and `.cssText` so that there's a consistent return type for the `css` tag and so the static style property has a single item type.

This would change below to just:

```ts
   const cssText = values.reduce((acc, v, idx) =>
       acc + literalValue(v) + strings[idx + 1], strings[0]);
   return new CSSResult(cssText);
```

Reply:
Done.